test/WebJobs.Script.Tests/Rpc/FunctionDispatcherTests.cs

```diff
@@ -46,6 +46,33 @@ public async void Starting_MultipleWebhostChannels_Succeeds()
             Assert.Equal(0, finalJobhostChannelCount);
         }
 
+        [Fact]
+        public async Task TerminateTests()
+        {
+            int expectedProcessCount = 2;
+            RpcFunctionInvocationDispatcher functionDispatcher = GetTestFunctionDispatcher(expectedProcessCount.ToString());
+            await functionDispatcher.InitializeAsync(GetTestFunctionsList(LanguageWorkerConstants.NodeLanguageWorkerName));
+
+            await WaitForJobhostWorkerChannelsToStartup(functionDispatcher, expectedProcessCount);
+
+            for (int restartCount = 0; restartCount < expectedProcessCount * 3; restartCount++)
+            {
+                foreach (var currChannel in functionDispatcher.JobHostLanguageWorkerChannelManager.GetChannels())
+                {
+                    var convertedChannel = (TestLanguageWorkerChannel)currChannel;
+                    convertedChannel.Cache.Add(1);
+                }
+            }
+            await functionDispatcher.TerminateAsync();
+            for (int restartCount = 0; restartCount < expectedProcessCount * 3; restartCount++)
+            {
+                foreach (var currChannel in functionDispatcher.JobHostLanguageWorkerChannelManager.GetChannels())
+                {
+                    Assert.True(((TestLanguageWorkerChannel)currChannel).Cache.Count == 0);
```

Comment:
you need to either use mocked version of ILanguageWorkerChannel or Implement Drain method on TestLangaugeWorkerChannel and verify invocations are drained

Reply:
functionDispatcher.ShutdownAsync() in turn invokes DrainAsync() on TestLanguageWorkerChannel which in turn does a executingInvocations.Clear(). This code already does what you mentioned unless I'm missing something else here.